### ISSUE TITLE
sporadic timeouts producing to multiple auto-created topics

### ISSUE THREAD

**Issue by aisch** [dpkp/kafka-python#539]
seeing this:
```
ERROR : kafka.cluster : Topic 5f2dc140e6e84fff8dc1acca1e186509 is not available during auto-create initialization
DEBUG : kafka.cluster : Updated cluster metadata version 3 to Cluster(brokers: 1, topics: 1, groups: 0)
```

looks like `KafkaProducer._wait_on_metadata` doesn't retry metadata request https://github.com/dpkp/kafka-python/blob/master/kafka/producer/kafka.py#L459-L468.

**Comment by dpkp** [comment]
The issue here is that the metadata response handler ignores errors when multiple topics are requested. It should work for single-topic producers. I will think a bit about how to manage state in the multiple topic case.

**Comment by dpkp** [comment]
Referred back to the java client and decided to simplify and stay consistent with the approach there. Fix is in master @ `bce0cad`

**Comment by aisch** [comment]
@dpkp :+1: verified fixed for my use case w/ 9f0db5d38b444f5a93da7bed4a19114aff8701e8
